testing/test_pytester.py

```diff
@@ -776,3 +776,28 @@ def test_makefile_joins_absolute_path(testdir: Testdir) -> None:
     else:
         p1 = testdir.makepyfile(**{str(absfile): ""})
         assert str(p1) == (testdir.tmpdir / absfile) + ".py"
+
+
+def test_testdir_created_files_none(testdir):
+    assert set() == testdir.created_files
+
+
+def test_testdir_duplicate_paths(testdir):
+    testdir.makepyfile("foo")
+    testdir.makepyfile("bar")
+    assert len(testdir.created_files) == 1
```

Comment:
```suggestion
    assert testdir.created_files == {testdir.tmpdir.join("test_testdir_duplicate_paths.py")}
```

This is better because we are more explicit about the outcome, helping to figure out any problems in case the test fails.